src/basic/parse-util.c

```diff
@@ -490,3 +490,40 @@ int safe_atod(const char *s, double *ret_d) {
         *ret_d = (double) d;
         return 0;
 }
+
+int parse_fractional_part_u(const char **p, size_t digits, unsigned *res) {
+        size_t i;
+        unsigned val = 0;
+        const char *s;
+
+        s = *p;
+
+        /* accept any number of digits, strtoull is limted to 19 */
```

Comment:
Well, `strtoll("333333333333333333333333333333333333333333")` is `errno=ERANGE` and I want to avoid copying the string.